packages/test/test-end-to-end-tests/src/test/gc/gcBlobInSummaries.spec.ts

```diff
@@ -0,0 +1,291 @@
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
+        });
+
+        it("GC blob handle written when summary fails", async () => {
+            // Make a change in dataStoreA.
+            dataStoreA._root.set("key", "value");
+
+            // A gc blob handle should be submitted as there are no gc changes
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            await submitFailingSummary(provider, summarizerClient1, logger, FailingSubmitSummaryStage.Generate);
+
+            // GC blob handle expected
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+        });
+
+        it("GC blob written when summary fails", async () => {
+            // Make a reference change by deleting a handle
+            dataStoreA._root.delete("dataStoreB");
+
+            await provider.ensureSynchronized();
+
+            await submitFailingSummary(provider, summarizerClient1, logger, FailingSubmitSummaryStage.Upload);
+
+            // GC blob expected as the summary had changed
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
+        });
+
+        it("GC blob handle written when new summarizer loaded from last summary summarizes", async () => {
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            await provider.ensureSynchronized();
+
+            // Make a reference change by deleting a handle
+            dataStoreA._root.delete("dataStoreB");
+
+            await submitFailingSummary(provider, summarizerClient1, logger, FailingSubmitSummaryStage.Generate);
+
+            // GC blob expected as the summary had changed
+            await submitSummaryAndValidateState(summarizerClient1, isBlob);
+
+            const summarizerClient2 = await getNewSummarizer();
+
+            // GC blob expected to be the same as the summary has not changed
+            await submitSummaryAndValidateState(summarizerClient2, isBlobHandle);
+        });
+
+        it("GC blob handle written when new summarizer loaded from older summary summarizes", async () => {
+            const summarizerClient2 = await getNewSummarizer();
+            await submitSummaryAndValidateState(summarizerClient1, isBlobHandle);
+
+            await provider.ensureSynchronized();
+
+            // Make a reference change by deleting a handle
+            dataStoreA._root.delete("dataStoreB");
+
+            await submitFailingSummary(provider, summarizerClient1, logger, FailingSubmitSummaryStage.Generate);
+
+            // GC blob expected as the summary had changed
+            const blob1 = await submitSummaryAndValidateState(summarizerClient1, isBlob);
+
+            // GC blob expected to be the same as the summary has not changed
+            const blob2 = await submitSummaryAndValidateState(summarizerClient2, isBlob);
+
+            assert.deepEqual(blob1, blob2, `Expected blobs to be equal!`);
```

Comment:
The assert fails here as the blobs are not equal. 
`const blob2 = await submitSummaryAndValidateState(summarizerClient2, isBlob);` should actually be `const blob2 = await submitSummaryAndValidateState(summarizerClient2, isBlobHandle);` as they should be equal. This is a race condition where `summarizerClient2` loads from an older summary before receiving the ack from `summarizerClient1`.